public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,879 @@
+  You can't call _setValue_ in `ngOnInit` or you'll miss the user's selections.
+  Put the _setValue_ logic in the [ngOnChanges](lifecyle-hooks.html#onchanges)
+  hook, which Angular calls whenever the input `hero` property changes.
+
+  You should also _reset the form_ when the hero changes so that 
+  control values from the previous hero are cleared and 
+  status flags are restored to the _pristine_ state.
+
+  Here's what `ngOnChanges` should look like:
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'set-value-on-changes','app/hero-detail.component.ts (excerpt)')(format=".")
+
+a#hero-list
+:marked
+  ### Create the _HeroListComponent_ and _HeroService_
+
+  The `HeroDetailComponent` is a nested sub-component of the `HeroListComponent` in a _master/detail_ view.
+  Together they look a bit like this:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/hero-list.png" width="420px" alt="HeroListComponent")
+:marked
+  The `HeroListComponent` uses an injected `HeroService` to retrieve heroes from the server
+  and then presents those heroes to the user as a series of buttons.
+  The `HeroService` emulates an HTTP service. 
+  It returns a `Promise` of heroes that resolves after a short delay, 
+  both to simulate network latency and to indicate visually 
+  the necessarily asynchronous nature of the application.
+
+  When the user clicks on a hero,
+  the component sets its `selectedHero` property which
+  is **bound to the `hero` input property** of the `HeroDetailComponent`.
+  The `HeroDetailComponent` detects the changed hero and re-sets its form
+  with that hero's data values.
+
+  A "Refresh" button clears the hero list and the current selected hero before refetching the heroes.
+  The `HeroListComponent` also has an `onSave` method that you'll wire up later in this guide.
+
+  The remaining `HeroListComponent` and `HeroService` implementation details are not relevant to understanding reactive forms.
+  The techniques involved are covered elsewhere in the documentation, including the _Tour of Heroes_
+  [here](../tutorial/toh-pt3.html "ToH: Multiple Components") and [here](../tutorial/toh-pt4.html "ToH: Services").
+
+  If you're coding along with the steps in this reactive forms tutorial, 
+  create the pertinent files based on the 
+  [source code displayed below](#source-code "Reactive Forms source code").
+  Then return here to learn about _form array_ properties.
+
+.l-main-section
+a#form-array
+:marked
+  ## Use _FormArray_ to present an array of _FormGroups_
+  So far, you've seen `FormControls` and `FormGroups`.
+  A `FormGroup` is a named object whose property values are `FormControls` and other `FormGroups`.
+
+  Sometimes you need to present an arbitrary number of controls or groups.
+  For example, a hero may have zero, one, or any number of addresses.
+  
+  Hero addresses aren't named. The `Hero.addresses` property is an array of `Address` instances.
```

Comment:
This does not seem to be needed here. Even if they were named, we'd still want to build a FormArray here.